code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -494,66 +369,78 @@
+/mob/living/simple_animal/slime/proc/slime_pre_attack(mob/living/simple_animal/slime/our_slime, atom/target, proximity, modifiers)
+	SIGNAL_HANDLER
+	if(isAI(target)) //The aI is not tasty!
+		to_chat(our_slime, "This being is incompatible, and not tasty.")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(our_slime.buckled == target) //If you try to attack the creature you are latched on, you instead cancel feeding
+		our_slime.stop_feeding()
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(iscyborg(target))
+		var/mob/living/silicon/robot/borg_target = target
+		borg_target.flash_act()
+		do_sparks(5, TRUE, borg_target)
+		var/stunprob = our_slime.powerlevel * 7 + 10
+		if(prob(stunprob) && our_slime.powerlevel >= 8)
+			our_slime.powerlevel -= 3
+			borg_target.adjustBruteLoss(our_slime.powerlevel * rand(6,10))
+			borg_target.visible_message(span_danger("The [our_slime.name] shocks [borg_target]!"), span_userdanger("The [our_slime.name] shocks you!"))
+		else
+			borg_target.visible_message(span_danger("The [our_slime.name] fails to hurt [borg_target]!"), span_userdanger("The [our_slime.name] failed to hurt you!"))
 
-///Removes everyone from the friend list
-/mob/living/simple_animal/slime/proc/clear_friends()
-	for(var/mob/friend as anything in Friends)
-		remove_friend(friend)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
 
-///The passed source will be no longer be the slime's target, leader, or one of its friends
-/mob/living/simple_animal/slime/proc/clear_memories_of(datum/source)
-	SIGNAL_HANDLER
-	if(source == Target)
-		set_target(null)
-	if(source == Leader)
-		set_leader(null)
-	remove_friend(source)
+	if(iscarbon(target) && our_slime.powerlevel > 0)
+		var/mob/living/carbon/carbon_target = target
+		var/stunprob = our_slime.powerlevel * 7 + 10  // 17 at level 1, 80 at level 10
+		if(!prob(stunprob))
+			return
 
-#undef SLIME_CARES_ABOUT
+		carbon_target.visible_message(span_danger("The [our_slime.name] shocks [carbon_target]!"), span_userdanger("The [our_slime.name] shocks you!"))
+
+		do_sparks(5, TRUE, carbon_target)
+		var/power = our_slime.powerlevel + rand(0,3)
+		carbon_target.Paralyze(power * 2 SECONDS)
+		carbon_target.set_stutter_if_lower(power * 2 SECONDS)
+		if (prob(stunprob) && our_slime.powerlevel >= 8)
+			carbon_target.adjustFireLoss(our_slime.powerlevel * rand(6,10))
+			carbon_target.updatehealth()
+
+	if(isslime(target))
+		if(target == our_slime)
+			return COMPONENT_CANCEL_ATTACK_CHAIN
+		var/mob/living/simple_animal/slime/target_slime = target
+		if(target_slime.buckled)
+			target_slime.stop_feeding(silent = TRUE)
+			visible_message(span_danger("[our_slime] pulls [target_slime] off!"), \
+				span_danger("You pull [target_slime] off!"))
+			return
+		target_slime.attacked_stacks += 5
+		if(target_slime.nutrition >= 100) //steal some nutrition. negval handled in life()
+			var/stolen_nutrition = our_slime.life_stage == SLIME_LIFE_STAGE_ADULT ? 90 : 50
```

Comment:
Suggestion: Just so this code is a bit more readable, extract this condition out and place it somewhere up the chain
`var/is_adult_slime = our_slime.life_stage == SLIME_LIFE_STAGE_ADULT`
`var/meme = is_adult_slime ? a : b`